cpp/src/gandiva/filter.cc

```diff
@@ -118,11 +135,15 @@ Status Filter::Make(SchemaPtr schema, ConditionPtr condition,
   // Return if the expression is invalid since we will not be able to process further.
   ExprValidator expr_validator(llvm_gen->types(), schema);
   ARROW_RETURN_NOT_OK(expr_validator.Validate(condition));
-  ARROW_RETURN_NOT_OK(llvm_gen->Build({condition}, SelectionVector::Mode::MODE_NONE));
+  ARROW_RETURN_NOT_OK(llvm_gen->Build({condition}, SelectionVector::Mode::MODE_NONE, obj_cache)); // to use when caching only the obj code
 
   // Instantiate the filter with the completely built llvm generator
   *filter = std::make_shared<Filter>(std::move(llvm_gen), schema, configuration);
-  cache.PutModule(cache_key, *filter);
+
+  filter->get()->SetCompiledFromCache(llvm_flag); // to use when caching only the obj code
+  used_cache_size_ = shared_cache->getCacheSize(); // track filter cache memory use
```

Comment:
When possible, put the comment one line above the code line it refers, that way is better to read than in front of the code

Reply:
Done